compression/gzip_zinfo.go

```diff
@@ -70,7 +73,7 @@ func (i *GzipZinfo) Close() {
 func (i *GzipZinfo) Bytes() ([]byte, error) {
 	blobSize := C.get_blob_size(i.cZinfo)
 	bytes := make([]byte, uint64(blobSize))
-	if bytes == nil {
+	if len(bytes) == 0 {
```

Comment:
Is this necessary? I think the Go runtime would panic if it can't allocate enough memory.

Reply:
if the passed `blobSize` is 0,`make` will return an _empty (not nil)_ slice. So `bytes` will pass the if check and panic later when `&bytes[0]` is called. Both nil/empty slices have 0-length.